### ISSUE TITLE
intermittent unit test failure: test/net_tests.cpp(311): error: in "net_tests/cnetaddr_basic": check addr_str == scoped_addr || addr_str == "fe80:0:0:0:0:0:0:1" has failed

### ISSUE THREAD

**Issue by MarcoFalke** [bitcoin/bitcoin#21682]
https://cirrus-ci.com/task/5470546704990208?logs=ci#L4527

**Comment by jonatack** [comment]
Thanks. Was about to push a fix after @ryanofsky reported on it lately.

**Comment by jonatack** [comment]
https://github.com/bitcoin/bitcoin/pull/19951#discussion_r611257547

**Comment by practicalswift** [comment]
I've looked at that test before and I don't think that specific `BOOST_CHECK` makes much sense TBH :)

1.) I don't understand why we test if `ToString()` output includes `%zone_id`: it clearly doesn't on some platforms, so we cannot rely on it anyways. Then why test it?

2.) And perhaps more fundamentally: why would we even _want_ to have `%zone_id` in our `ToString()` output? I think the expectation is to get say `2001:0db8:85a3::8a2e:0370:7334 ` and not `2001:0db8:85a3::8a2e:0370:7334%eth2` when doing `ipv6_addr.ToString()` :)

Suggested solution:

```diff
diff --git a/src/test/net_tests.cpp b/src/test/net_tests.cpp
index 8eab26f3d..61cb381aa 100644
--- a/src/test/net_tests.cpp
+++ b/src/test/net_tests.cpp
@@ -307,10 +307,6 @@ BOOST_AUTO_TEST_CASE(cnetaddr_basic)
     BOOST_REQUIRE(addr.IsValid());
     BOOST_REQUIRE(addr.IsIPv6());
     BOOST_CHECK(!addr.IsBindAny());
-    const std::string addr_str{addr.ToString()};
-    BOOST_CHECK(addr_str == scoped_addr || addr_str == "fe80:0:0:0:0:0:0:1");
-    // The fallback case "fe80:0:0:0:0:0:0:1" is needed for macOS 10.14/10.15 and (probably) later.
-    // Test that the delimiter "%" and default zone id of 0 can be omitted for the default scope.
     BOOST_REQUIRE(LookupHost(link_local + "%0", addr, false));
     BOOST_REQUIRE(addr.IsValid());
     BOOST_REQUIRE(addr.IsIPv6());
```

:)

**Comment by jonatack** [comment]
Noting here that this was also fixed by #21690 as I wrote above in https://github.com/bitcoin/bitcoin/issues/21682#issuecomment-819735520 that I would do. #21690 not only fixed the issue but also kept the test coverage and improved the contextual information to avoid the confusion above and that I think was worth keeping.
